src/components/Table/GridTable.test.tsx

```diff
@@ -1036,6 +1045,48 @@ describe("GridTable", () => {
     localStorage.setItem(tableIdentifier, "");
   });
 
+  it("can select all", async () => {
+    // Given a parent with a child and grandchild
+    const rows: GridDataRow<NestedRow>[] = [
+      { kind: "header", id: "header" },
+      {
+        ...{ kind: "parent", id: "p1", name: "parent 1" },
+        children: [
+          {
+            ...{ kind: "child", id: "p1c1", name: "child p1c1" },
+            children: [{ kind: "grandChild", id: "p1c1g1", name: "grandchild p1c1g1" }],
+          },
+        ],
+      },
+    ];
+    const api: MutableRefObject<GridTableApi | undefined> = { current: undefined };
```

Comment:
In production code, this `api` prop would be a `useRef` that the page passes into `GridTable` to "get back" a pointer to API methods it can call to ask/interrogate `GridTable` about various things, i.e. what is your current list of selected rows.

Note that I don't have an `onRowsSelected` callback yet...should be easy to add...will probably do as a separate PR...